src/devtools/DevToolsViewProvider.ts

```diff
@@ -0,0 +1,129 @@
+      localResourceRoots: [this._extensionUri],
+    };
+    panel.webview.html = DevToolsViewProvider._getHtmlForWebview(
+      panel.webview,
+      this._extensionUri,
+    );
+
+    panel.webview.onDidReceiveMessage((data) => {
+      devtoolsEvents.emit("fromDevTools", data);
+    });
+    function forwardToDevTools(data: unknown) {
+      panel.webview.postMessage(data);
+    }
+    devtoolsEvents.addListener("toDevTools", forwardToDevTools);
+    panel.onDidDispose(() => {
+      devtoolsEvents.removeListener("toDevTools", forwardToDevTools);
+    });
+  }
+
+  private static _getHtmlForWebview(
+    webview: vscode.Webview,
+    extensionUri: vscode.Uri,
+  ) {
+    // Get the local path to main script run in the webview, then convert it to a uri we can use in the webview.
+    const scriptUri = webview.asWebviewUri(
+      vscode.Uri.joinPath(extensionUri, "lib", "panel.js"),
+    );
+
+    // Use a nonce to only allow a specific script to be run.
+    const nonce = getNonce();
+
+    return `
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+
+    <!--
+					Use a content security policy to only allow loading images from https or from our extension directory,
+					and only allow scripts that have a specific nonce.
+		-->
+		<meta http-equiv="Content-Security-Policy" content="
+      default-src 'none';
+      style-src ${webview.cspSource} https://fonts.googleapis.com 'unsafe-inline';
+      font-src ${webview.cspSource} https://fonts.gstatic.com;
+      img-src ${webview.cspSource} https:;
+      connect-src ${webview.cspSource} https://*.github.com;
+      script-src 'nonce-${nonce}';
+      frame-src https://*.apollographql.com/;
+    ">
+				<meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+    <title>Apollo Client DevTools</title>
+    <style>
+      html,
+      body {
+        overflow: hidden;
+        font-size: var(--vscode-font-size);
+      }
+      ::-webkit-scrollbar {
+        display: none;
+      }
+      #devtools {
+        width: 100vw;
+        height: 100vh;
+      }
+    </style>
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Fira+Code&family=Inter:wght@400;500;600;700&display=swap"
+      rel="stylesheet"
+    />
+  </head>
+  <body class="text-primary dark:text-primary-dark">
+    <div id="devtools"></div>
+    <script nonce="${nonce}">
+      const vscode = acquireVsCodeApi();
+      try {
+        // remove VSCode default styles
+        _defaultStyles.remove();
+      } catch {}
+      window.originalPostMessage = window.postMessage;
+      window.postMessage = function wrapPostMessage (...args) {
+        if (args.length>1 && args[1].startsWith("vscode-webview://")) {
+          return window.originalPostMessage(...args);
+        }
+        return vscode.postMessage(...args);
+      };
+
+      window.addEventListener("message", (event) => {  console.debug(event); });
```

Comment:
```suggestion
      window.addEventListener("message", (event) => { console.debug(event); });
```

Sometimes I wish my obsession with spacing didn't kick in 🤣

Reply:
Sometimes I wish my prettier would, though. 🙈